app/src/androidTest/java/com/pileproject/drive/execution/NxtControllerTest.java

```diff
@@ -0,0 +1,214 @@
+        assertEquals(controller.getSoundSensorValue(), -1);
+    }
+
+    @Test
+    public void whenSoundSensorIsNotNull_thenReturnProperValue() throws Exception {
+        Whitebox.setInternalState(controller, "mSoundSensor", soundSensor);
+
+        doReturn(10).when(soundSensor).getDb();
+        assertEquals(10, controller.getSoundSensorValue());
+
+        doReturn(100).when(soundSensor).getDb();
+        assertEquals(100, controller.getSoundSensorValue());
+
+        doReturn(-10).when(soundSensor).getDb();
+        assertEquals(-10, controller.getSoundSensorValue());
+    }
+
+    @Test
+    public void whenLineSensorIsNull_thenReturnNegative() throws Exception {
+        assertEquals(controller.getLineSensorValue(), -1);
+    }
+
+    @Test
+    public void whenLineSensorIsNotNull_thenReturnProperValue() throws Exception {
+        Whitebox.setInternalState(controller, "mLineSensor", lineSensor);
+
+        doReturn(20).when(lineSensor).getSensorValue();
+        assertEquals(20, controller.getLineSensorValue());
+
+        doReturn(200).when(lineSensor).getSensorValue();
+        assertEquals(200, controller.getLineSensorValue());
+
+        doReturn(-20).when(lineSensor).getSensorValue();
+        assertEquals(-20, controller.getLineSensorValue());
+    }
+
+    @Test
+    public void testMoveForward() throws Exception {
+        setUpMotors();
+
+        controller.moveForward();
+
+        verify(leftMotor).forward();
+        verify(rightMotor).forward();
+    }
+
+    @Test
+    public void testMoveBackward() throws Exception {
+        setUpMotors();
+
+        controller.moveBackward();
+
+        verify(leftMotor).backward();
+        verify(rightMotor).backward();
+    }
+
+    @Test
+    public void testTurnLeft() throws Exception {
+        setUpMotors();
+
+        controller.turnLeft();
+
+        verify(leftMotor).backward();
+        verify(rightMotor).forward();
+    }
+
+    @Test
+    public void testTurnRight() throws Exception {
+        setUpMotors();
+
+        controller.turnRight();
+
+        verify(leftMotor).forward();
+        verify(rightMotor).backward();
+    }
+
+    @Test
+    public void testHalt() throws Exception {
+        setUpMotors();
+
+        controller.halt();
+
+        verify(leftMotor).stop();
+        verify(rightMotor).stop();
+    }
+
+    @Test
+    public void testSetMotorPower() throws Exception {
+        setUpMotors();
+
+        // at first, they have not been initialized
+        assertEquals(0, leftMotor.getSpeed());
+        assertEquals(0, rightMotor.getSpeed());
+
+        controller.moveForward();
+
+        // after a move command, they will be initialized with the default values
+        assertEquals(NxtController.INIT_MOTOR_POWER, leftMotor.getSpeed());
+        assertEquals(NxtController.INIT_MOTOR_POWER, rightMotor.getSpeed());
+
+        controller.setMotorPower(NxtController.MotorKind.LeftMotor, 100);
```

Comment:
It's better to split into two methods:
``whenMotorSpeedsAreNotInitialized_thenMovesForwardWithDefaultValue`` and ``whenSetMotorPowerCalled_thenMovesForwardWithTheValue``